### ISSUE TITLE
Generation of Constructor for HostServices-Service not working

### ISSUE THREAD

**Issue by angerhard** [GeeWee/boilerplatefree#18]
### Describe the bug

As a java developer, I want to have lombok feeling in c# .NET applications, so I decided to have boilerplatefree in my applications.

When I integrate boilerplatefree, the attribute "AutoGenerateConstructor" won't work with transient services. The injected dependencies are not initiated, so when using these dependencies, I receive a System.NullReferenceException.

Code Examples:

```csharp
[AutoGenerateConstructor]
    public partial class TestService
    {
        private readonly TestRepository _testRepository;
        private readonly Logger<TestService> _logger;

        public void runService()
        {
            var testData = _testRepository.getTestData();
            _logger.LogInformation("giving out testdata: {}", testData);
        }
    }
```

When calling this service, my expectation is that _testRepository has been initiated by AutoGenerateConstructor. 

### Steps to reproduce

Check out Project: https://github.com/angerhard/BoilerPlateFreeTesting
Start Program.cs

### Expected behaviour

Console should write out

giving out testData: some-test-data

**Comment by angerhard** [comment]
Some further testing: 
![image](https://user-images.githubusercontent.com/6179043/139851132-882055ea-59b4-483e-9fe3-51d0791c8cad.png)
When I construct the class manually, the IDE shows and validate the auto generated constructor.
When I execute this code, dotnet complains about the missing constructor:
(english translation): "TestService" contains no constructor with 2 arguments.

**Comment by GeeWee** [comment]
I think this should be solved by #19 - if you remove the `AutoGenerateConstructor` from your class without any fields, does it start working?

**Comment by angerhard** [comment]
I removed the Attribute from TestRepository (no fields to inject) and updated BPF to 1.3.1, but the error occurs again (TestRepo is empty in TestService)
